src/containers/cooperation-details/add-course-modal-modal/AddCourseTemplateModal.tsx

```diff
@@ -64,11 +69,73 @@ const AddCourseTemplateModal: FC<AddCourseTemplateModalProps> = ({
     setSearchValue('')
   }
 
+  const onCategoryChange = (
+    _: SyntheticEvent,
+    value: CategoryNameInterface | null
+  ) => {
+    updateFilterInQuery(value?._id ?? '', 'category')
+    updateFilterInQuery('', 'subject')
+  }
+
+  const onSubjectChange = (
+    _: SyntheticEvent,
+    value: SubjectNameInterface | null
+  ) => {
+    updateFilterInQuery(value?._id ?? '', 'subject')
+  }
+
+  const onLevelChange = (e: SelectChangeEvent<ProficiencyLevelEnum[]>) => {
+    const selectedProficiencyLevels = e.target.value as ProficiencyLevelEnum[]
+    updateFilterInQuery(selectedProficiencyLevels, 'proficiencyLevel')
+  }
+
   const onCreateCourse = () => {
     closeModal()
     navigate(authRoutes.myCourses.newCourse.path)
   }
 
+  const onModalCancel = () => {
+    closeModal()
+    resetFilters()
+  }
+
+  const toggleFilters = () => {
+    setShowFilters((prevShowFilters) => !prevShowFilters)
+  }
+
+  const filterBySearch = (item: Course): boolean =>
+    'title' in item &&
+    item.title.toLowerCase().includes(searchValue.toLowerCase())
+
+  const filterByCategory = (item: Course): boolean =>
+    !filters.category || item.category?._id === filters.category
+
+  const filterBySubject = (item: Course): boolean =>
+    !filters.subject || item.subject?._id === filters.subject
+
+  const filterByProficiencyLevel = (item: Course): boolean =>
+    !filters.proficiencyLevel.length ||
+    filters.proficiencyLevel.some(
+      (level) => item.proficiencyLevel?.includes(level)
+    )
+
+  const getItems = () => {
+    return response.items
+      .filter(
+        (item) =>
+          filterBySearch(item) &&
+          filterByCategory(item) &&
+          filterBySubject(item) &&
+          filterByProficiencyLevel(item)
```

Comment:
maybe it will be better to do all filters in one function and all only it here instead of 4 functions?

Reply:
done)